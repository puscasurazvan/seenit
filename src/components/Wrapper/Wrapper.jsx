import React, { useState, lazy, Suspense } from 'react'

import Heading from '../Heading'
import VideoModal from '../VideoModal'

import './Wrapper.scss'

const EmbeddedVideo = lazy(() => import('../EmbeddedVideo'))

const VideoWrapper = () => {
  const [show, setShow] = useState(false)

  const toggleVideo = () => {
    setShow(!show)
  }

  const videoData = {
    title: 'Seenit Camera Confidence Video',
    src: 'https://youtube.com/embed/OMMY3I1PjoI',
  }

  const displayText = 'How to overcome camera shyness'

  return (
    <div id="videoWrapper" className="VideoWrapper">
      <Heading
        headingText={displayText}
        toggleVideo={toggleVideo}
        modalOpen={show}
      />
      {show && (
        <VideoModal open={show} toggle={toggleVideo} buttonText={displayText}>
          <Suspense fallback={'Loading...'}>
            <EmbeddedVideo video={videoData} />
          </Suspense>
        </VideoModal>
      )}
    </div>
  )
}

export default VideoWrapper
