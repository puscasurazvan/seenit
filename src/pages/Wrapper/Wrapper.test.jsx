import React from 'react'
import { mount } from 'enzyme'
import Wrapper from './Wrapper'

describe('Wrapper', () => {
  it('should render with a Heading', () => {
    const wrapper = mount(<Wrapper />)
    expect(wrapper.find('Heading').length).toEqual(1)
  })

  it('should render a VideoModal when Heading is clicked', () => {
    const wrapper = mount(<Wrapper />)
    const heading = wrapper.find('h3')
    heading.simulate('click')
    expect(wrapper.find('VideoModal').length).toEqual(1)
  })

  it('should render without exploding', () => {
    const wrapper = mount(<Wrapper />)
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
