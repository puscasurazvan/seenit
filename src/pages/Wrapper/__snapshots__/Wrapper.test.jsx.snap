// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render without exploding 1`] = `ReactWrapper {}`;
